apps/test/unit/studio/itemTest.js

```diff
@@ -11,32 +16,79 @@ import {
   BEHAVIOR_GRID_ALIGNED
 } from '@cdo/apps/studio/constants';
 
-import Item from '@cdo/apps/studio/Item';
-
 describe('item', () => {
   before(() => {
     Studio.trackedBehavior = {};
     Studio.trackedBehavior.createdItems = {};
+    Studio.SQUARE_SIZE = 50;
   });
 
   let item;
 
   describe('update', () => {
+    describe('if destination is not yet set', () => {
+      before(() => {
+        // the destination-setting logic uses _.shuffle to semi-randomize the
+        // set of possible destinations. Stub that with something deterministic
+        // to make testing easier
+        sinon.stub(_, 'shuffle').callsFake(ar => ar);
+      });
+
+      let targetSprite;
+      beforeEach(() => {
+        item = new Item({});
+        item.x = 100;
+        item.y = 100;
+
+        item.hasWall = sinon.stub().returns(false);
+
+        targetSprite = new Sprite({});
+        targetSprite.x = 200;
+        targetSprite.y = 200;
+        Studio.sprite = [targetSprite];
+        item.targetSpriteIndex = 0;
+      });
+
+      it('sets an arbitrary direction on wander', () => {
+        item.activity = BEHAVIOR_WANDER;
+        item.update();
+        expect(item.dir).to.equal(Direction.NORTH);
```

Comment:
This ends up feeling a little funky, since the content of the stuff passed to shuffle is determined by a bunch of other stuff within the update logic.

What if I instead keep on overriding shuffle for each one like I'm doing, but also spy on it with some explanation?

```javascript
      // the destination-setting logic uses _.shuffle to semi-randomize the
      // set of possible destinations before sorting them by score. We would
      // instead like to make it deterministic.
      let shuffleSpy;

      beforeEach(() => {
        shuffleSpy = sinon.stub(_, 'shuffle').callsFake(ar => ar);
      });

      afterEach(() => {
        _.shuffle.restore();
      });

      ...

      it('sets an arbitrary direction on wander', () => {
        item.activity = BEHAVIOR_WANDER;

        // Since in the wander case all destinations are possible and all have a
        // score of one, whichever destination happens to be returned first will
        // be the coordinates to which we are headed. Since North happens to be
        // the first direction considered, North will be our final result.
        item.update();
        expect(shuffleSpy.callCount).to.equal(1);

        const firstDestination = shuffleSpy.firstCall.args[0][0];
        expect(firstDestination.gridX).to.equal(2);
        expect(firstDestination.gridY).to.equal(1);

        expect(item.dir).to.equal(Direction.NORTH);
        expect(item.destGridX).to.equal(2);
        expect(item.destGridY).to.equal(1);
      });
```